File: thesum.py
import os

sum=0
value=0
filename='sum.dat'

if not os.path.isfile(filename):
    print "Impossibile aprire il file!"
    sum=0
else:
    f=open(filename,'r')
    print "\nLeggo il file sum.dat ..."
    line=f.readline()
    somma=int(line)
    f.close()
    
print "\nIl valore letto e' = ",sum
value = input("\nInserisci il valore da sommare = ")
sum=sum+value
print "\nla somma e' = ",sum
print "\nSalvo il file con il nuovo valore ...",sum
f=open(filename,'w')
f.write(str(somma))
f.close()

File: thesum.c
#include <stdio.h>
int main()
{
    FILE *f;
    int sum=0, value=0;
    
    f=fopen("sum.dat","r");
    if(!f)
    {
        printf("\nImpossibile aprire sum.dat, il file non esiste!\n");
        sum=0;
    }
    else
    {
        printf("\nLeggo nel file sum.dat ...");
        fscanf(f,"%d",&sum);
        fclose(f);
    }
    
    printf("\nValore letto dal file = %d",sum);
    
    printf("\nInserisci un intero da sommare = ");
    scanf("%d",&value);
    
    sum=sum+value;
    printf("\nSomma = %d",sum);
    
    printf("\n0Salvo il valore nel file sum.dat ...\n");
    f=fopen("sum.dat","w");
    fprintf(f,"%d\n",sum);
    fclose(f);
    
    return 0;
}
